drivers/bmx280/bmx280.c

```diff
@@ -166,218 +394,39 @@ uint32_t bmx280_read_pressure(const bmx280_t *dev)
     return p_acc >> 8;
 }
 
-#if defined(MODULE_BME280)
+#if defined(MODULE_BME280_SPI) || defined(MODULE_BME280_I2C)
 uint16_t bme280_read_humidity(const bmx280_t *dev)
 {
+    assert(dev);
+
     const bmx280_calibration_t *cal = &dev->calibration; /* helper variable */
 
     /* Read the uncompensated pressure */
-    int32_t adc_H = (((uint32_t)measurement_regs[6 + 0]) << 8) |
-        (((uint32_t)measurement_regs[6 + 1]));
+    int32_t adc_H = (((uint32_t)RAW_DATA[6 + 0]) << 8) |
+        (((uint32_t)RAW_DATA[6 + 1]));
 
     /*
      * Compensate the humidity value.
-     * The following is code from Bosch's BME280_driver bme280_compensate_humidity_int32()
-     * The variable names and the many defines have been modified to make the code
-     * more readable.
+     * The following is code from Bosch's BME280_driver
+     * bme280_compensate_humidity_int32(). The variable names and the many
+     * defines have been modified to make the code more readable.
      * The value is first computed as a value in %rH as unsigned 32bit integer
      * in Q22.10 format(22 integer 10 fractional bits).
      */
     int32_t var1;
 
     /* calculate x1*/
-    var1 = (t_fine - ((int32_t)76800));
+    var1 = (dev->t_fine - ((int32_t)76800));
     /* calculate x1*/
     var1 = (((((adc_H << 14) - (((int32_t)cal->dig_H4) << 20) - (((int32_t)cal->dig_H5) * var1)) +
               ((int32_t)16384)) >> 15) *
             (((((((var1 * ((int32_t)cal->dig_H6)) >> 10) *
                  (((var1 * ((int32_t)cal->dig_H3)) >> 11) + ((int32_t)32768))) >> 10) +
                ((int32_t)2097152)) * ((int32_t)cal->dig_H2) + 8192) >> 14));
     var1 = (var1 - (((((var1 >> 15) * (var1 >> 15)) >> 7) * ((int32_t)cal->dig_H1)) >> 4));
-    var1 = (var1 < 0 ? 0 : var1);
-    var1 = (var1 > 419430400 ? 419430400 : var1);
+    var1 = (var1 < 0) ? 0 : var1;
+    var1 = (var1 > 419430400) ? 419430400 : var1;
```

Comment:
You don't happen to know the source of those magic values?

Reply:
nope. This code was already part of the driver before I touched it. It is magic code that is given by Bosch if I am not mistaken...